Kernel/asm/interrupts.asm

```diff
@@ -104,21 +105,37 @@ picMasterMask:
 	push rbp
     mov rbp, rsp
     mov ax, di
-    out	21h,al
+    out	21h, al
     pop rbp
     retn
 
 picSlaveMask:
-	push    rbp
-    mov     rbp, rsp
-    mov     ax, di  ; ax = mascara de 16 bits
-    out	0A1h,al
-    pop     rbp
+	push rbp
+    mov rbp, rsp
+    mov ax, di  ; ax is a 16 bits mask
+    out	0A1h, al
+    pop rbp
     retn
 
 ; 8254 Timer (Timer Tick)
 _irq00Handler:
-	irqHandlerMaster 0
+	push rsp
+	pushState
+
+	mov rdi, 0
+	call irqDispatcher
+	
+	mov rdi, rsp
+	call sch_switchProcess
+	mov rsp, rax
```

Comment:
Might not be the best idea to handle it like this; this would mean `sch_switchProcess()` would have to modify the stack (so once control comes back to this line, it can then popState to get the registers and execute whatever corresponding process) but also change the rsp.

This would also not allow us to reuse that function to run a process in any other case that's not an interrupt (e.g. new process just started). Maybe we should instead implement a `noreturn void sch_runNext()` (name it whatever) that chooses a process to run, and yields the CPU to it, and only returns when an error occurred.

Keep in mind that if calling it from an interrupt, you still need to signal EOI to the pic!

Reply:
TL;DR I don't think you should rely on modifying the stack. Syscall handlers also don't do pushState/popState (although we don't need to preserve all registers in those cases 😉 (only if we interrupt mid-syscall))